web-ui/src/pages/PendingSkillsPage.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useContext, useEffect, useState } from "react";
+
+import { AppContext, UPDATE_SKILLS } from "../context/AppContext";
+import { getPendingSkills } from "../api/skill";
+import PendingSkillsCard from "../components/pending_skills/PendingSkillsCard";
+import PendingSkillsModal from "../components/pending_skills/PendingSkillsModal";
+
+import { Button } from "@material-ui/core";
+
+import "./PendingSkillsPage.css";
+
+const PendingSkillsPage = () => {
+  const { state, dispatch } = useContext(AppContext);
+  const { csrf, skills } = state;
+
+  const [allSkills, setSkills] = useState(skills);
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = () => setOpen(true);
+
+  const handleClose = () => setOpen(false);
+
+  useEffect(() => {
```

Comment:
Instead of keeping a copy of the skills around in state and duplicating them for update, you should just issue an update of all skills when they come to this page and create a selector for pending skills. That removes all the complexity of the local state (none is needed here other than the open/closed state), as well as the complexity in the useEffect.